src/experiment_generator/om2_forcing_updater.py

```diff
@@ -0,0 +1,119 @@
+
+        write_json(file_read, forcing_path)
+
+    @staticmethod
+    def _find_matching_param_index(inputs: list, fieldname: str) -> int | None:
+        """
+        Locate the index of a parameter in the 'inputs' list by field name.
+        """
+        for i, base in enumerate(inputs):
+            if base.get("fieldname") != fieldname:
+                continue
+            return i
+        return None
+
+    def _preprocess_perturbations(self, fieldname: str, updates: dict) -> None:
+        """
+        process `updates["perturbations"]`.
+        Warns and removes the key from `updates` if unsuitable.
+        """
+        perts = updates.get("perturbations")
+
+        # treat falsy as "no change"
+        if perts in (None, {}, []):
+            warnings.warn(
+                f"-- forcing.json '{fieldname}': empty/None 'perturbations' provided; skipping.",
+                UserWarning,
+            )
+            updates.pop("perturbations", None)
+            return
+
+        # accept dict -> wrap it to list[dict]
+        if isinstance(perts, dict):
+            perts = [perts]
+            updates["perturbations"] = perts
+
+        # accept list[dict]
+        elif isinstance(perts, list) and all(isinstance(pert, dict) for pert in perts):
+            pass
+        else:
+            raise TypeError(f"-- forcing.json '{fieldname}': 'perturbations' must be a dict or list of dicts")
+
+        required = ["type", "dimension", "value", "calendar", "comment"]
+
+        def _is_to_remove(pert: dict) -> bool:
+            if all(pert.get(k) in REMOVED for k in required):
+                return True
+            if pert.get("type") in REMOVED:
+                return True
+            return False
+
+        cleaned = [p for p in perts if not _is_to_remove(p)]
+        if not cleaned:
+            warnings.warn(
+                f"-- forcing.json: all perturbations for field '{fieldname}' are marked for removal; "
+                "skipping perturbations.",
+                UserWarning,
+            )
+            # Drop the key so it doesn't carry an empty/removed set forward
+            updates.pop("perturbations", None)
+            return
+
+        # validate each dict
+        for pert in cleaned:
+            self._validate_single_perturbation(pert)
+
+        # keep only the cleaned ones
+        updates["perturbations"] = cleaned
+
+    @staticmethod
+    def _validate_single_perturbation(pert: dict) -> None:
+        """
+        Validate a single perturbation dict.
+        """
+        if pert["type"] not in {"scaling", "offset", "separable"}:
+            raise ValueError(f"Invalid perturbation type: {pert['type']}")
+
+        dim = pert["dimension"]
+        accepted_dim = (isinstance(dim, str) and dim in {"spatial", "temporal", "constant", "spatiotemporal"}) or (
+            isinstance(dim, list) and (dim == ["temporal", "spatial"] or dim == ["spatial", "temporal"])
```

Comment:
```suggestion
            isinstance(dim, list) and (dim == ["temporal", "spatial"] or dim == ["spatial", "temporal"])
```
I seems from here that only `["temporal", "spatial"]` is valid: https://github.com/COSIMA/libaccessom2/blob/master/libforcing/src/forcing_config.F90#L217-L240